### ISSUE TITLE
allow to change the representation of the subject

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium-roadmap#18]
- add option to Expect
- add option to feature

keep in mind that more Options will most likely follow for Expect. Things like, use a different reporter, use a coloured output in the console etc

**Comment by robstoll** [comment]
@jGleitz what are your thoughts about the following syntax.
```
expect(person.name, representation = RawString.create("Person[firstName=${person.name}, ...]"))
  .toBe("peter")
```
resulting in 
```
expect: Person[firstName=robert, ...]
* to be: peter
```

Other options such as use a different Reporter/nullRepresentation etc. would be configured via ExpectOptions:
```
expect(nullableInt, ExpectOptions(nullRepresentation = "<NULL>"))
  .toBe(1)
```
resulting in 
```
expect: <NULL>
* to be: 1
```

Where also a builder can be used to benefit of things like, set a verb without the need to wrap it into an Untranslatable etc.
```
expect(1, ExpectOptions {
     withVerb("I assert that")
     withNullRepresentation("<NULL>")
     withReporter(ReporterBuilder...)
}).toBe(2)
```
Of course, one would probably want to hide the last version behind an own assertion verb, something along the line of:
```
fun <T> expect(t: T, representation: Any? = null) = 
  ch.tutteli.atrium.verbs.api.expect(t, ExpectOptions {
     withVerb("I assert that")
     withNullRepresentation("<NULL>")
     withReporter(ReporterBuilder...)
 })
```

**Comment by jGleitz** [comment]
That proposal looks very powerful and expressive! :+1: 

I would suggest one change: Do not force the user to use `RawString`/`Translatable`/`Text` or any of the like. These types solve problems that the user should not need to care about at the API level!

So I would suggest that the `representation` can also be a string. Whether there is still a usecase to also offer a shortcut with `RawString` or `Reportable` can be debated. My intuition is that a `String` `representation` is enough because everything else can be doen through `ExpectOptions`.

**Comment by robstoll** [comment]
fine with me, let's start with String and withRepresentation which accepts Any?

**Comment by robstoll** [comment]
closing it as it will be resolved by robstoll/atrium#153
